ocs_ci/ocs/longevity.py

```diff
@@ -0,0 +1,711 @@
+        self.pvc_size = None
+        self.pvc_count = None
+        self.pod_count = None
+        self.num_of_obcs = None
+
+    def construct_stage_builder_bulk_pvc_creation_yaml(self, num_of_pvcs, pvc_size):
+        """
+        This function constructs pvc.yamls to create bulk of pvc's using kube_jobs.
+        It constructs yamls with the specified number of PVCs of each supported
+        type and access mode.
+
+        Eg: num_of_pvcs = 30
+        The function creates a total of 120 PVCs of below types and access modes
+        RBD-Filesystemvolume  -> 30 RWO PVCs
+        CEPHFS                -> 30 RWO PVCs, 30 RWX PVCs
+        RBD-Block             -> 30 RWX PVCs
+
+        Args:
+            num_of_pvcs(int): Bulk PVC count
+            pvc_size (str): size of all pvcs to be created with Gi suffix (e.g. 10Gi).
+            If None, random size pvc will be created
+
+        Returns:
+             pvc_dict_list (list): List of all PVC.yaml dicts
+
+        """
+        pvc_dict_list = []
+        # Construct bulk pvc creation yaml for ceph-rbd with access mode RWO
+        log.info(
+            "Constructing bulk pvc creation yaml for ceph-rbd with access mode RWO"
+        )
+        pvc_dict_list1 = construct_pvc_creation_yaml_bulk_for_kube_job(
+            no_of_pvc=num_of_pvcs,
+            access_mode=constants.ACCESS_MODE_RWO,
+            sc_name=constants.CEPHBLOCKPOOL_SC,
+            pvc_size=pvc_size,
+        )
+        pvc_dict_list.append(pvc_dict_list1)
+        # Construct bulk pvc creation yaml for ceph-rbd with access mode RWX
+        log.info(
+            "Constructing bulk pvc creation yaml for ceph-rbd with access mode RWX"
+        )
+        pvc_dict_list2 = construct_pvc_creation_yaml_bulk_for_kube_job(
+            no_of_pvc=num_of_pvcs,
+            access_mode=constants.ACCESS_MODE_RWX,
+            sc_name=constants.CEPHBLOCKPOOL_SC,
+            pvc_size=pvc_size,
+        )
+        pvc_dict_list.append(pvc_dict_list2)
+        # Construct bulk pvc creation yaml for cephfs with access mode RWO
+        log.info("Constructing bulk pvc creation yaml for cephfs with access mode RWO")
+        pvc_dict_list3 = construct_pvc_creation_yaml_bulk_for_kube_job(
+            no_of_pvc=num_of_pvcs,
+            access_mode=constants.ACCESS_MODE_RWO,
+            sc_name=constants.CEPHFILESYSTEM_SC,
+            pvc_size=pvc_size,
+        )
+        pvc_dict_list.append(pvc_dict_list3)
+        # Construct bulk pvc creation yaml for cephfs with access mode RWX
+        log.info("Constructing bulk pvc creation yaml for cephfs with access mode RWX")
+        pvc_dict_list4 = construct_pvc_creation_yaml_bulk_for_kube_job(
+            no_of_pvc=num_of_pvcs,
+            access_mode=constants.ACCESS_MODE_RWX,
+            sc_name=constants.CEPHFILESYSTEM_SC,
+            pvc_size=pvc_size,
+        )
+        pvc_dict_list.append(pvc_dict_list4)
+
+        return pvc_dict_list
```

Comment:
instead of appending and returning it as whole will it be good to return individual list ?
es: return rbd_rwo, cephfs_rwx etc...

Reply:
I am using this all_pvc_dict_list to construct stagebuilder kube job, so I had to append all these to a list